easybuild/easyconfigs/a/Amber/Amber-18-AT-18_use_fftw_from_mkl_or_external.patch

```diff
@@ -0,0 +1,158 @@
+Force using FFTW from external or mkl.
+
+Åke Sandgren, 20170517
+diff -ru amber18.orig/AmberTools/src/configure2 amber18/AmberTools/src/configure2
+--- amber18.orig/AmberTools/src/configure2	2018-04-16 01:39:34.000000000 +0200
++++ amber18/AmberTools/src/configure2	2018-10-30 09:46:16.005558625 +0100
+@@ -3166,63 +3166,83 @@
+ fi
+ 
+ if [ "$has_fftw3" = 'yes' ]; then
+-    echo
+-    echo "Configuring fftw-3.3 (may be time-consuming)..."
+-    echo
+-    enable_mpi=""
+-    enable_debug=""
+-    enable_sse="--enable-sse=no --enable-sse2=no --enable-avx=no"
+-    mpicc=""
+-    if [ "$mpi" = "yes" ]; then
+-        enable_mpi="--enable-mpi=yes"
+-    fi
+-    if [ "$intelmpi" = "yes" ]; then
+-        mpicc="MPICC=mpiicc"
+-    fi
+-    if [ "$debug" = "yes" ]; then
+-        enable_debug="--enable-debug=yes --enable-debug-malloc=yes --enable-debug-alignment=yes"
+-    fi
+-    if [ "$sse" = "yes" ]; then
+-        enable_sse="--enable-sse2=yes" # --enable-avx=yes"
+-    fi
+-    if [ "$mic" = 'yes' ]; then
+-      echo "   --configuring for mic (native mode)..."
+-      echo
+-      cd fftw-3.3 && \
+-      ./configure --prefix=$amberprefix --libdir=$amberprefix/lib \
+-        --enable-static --enable-shared --host=x86_64-k1om-linux \
+-      --build=x86_64-unknown-linux \
+-       $enable_mpi $mpicc $enable_debug \
+-        CC="$cc -mmic" CFLAGS="$cflags $coptflags " \
+-        F77="$fc -mmic" FFLAGS="$fflags $foptflags " \
+-        FLIBS="$flibs_arch" \
+-        > ../fftw3_config.log 2>&1
+-      ncerror=$?
+-    else
+-      cd fftw-3.3 && \
+-        ./configure --prefix=$amberprefix --libdir=$amberprefix/lib \
+-        --enable-static --enable-shared \
+-        $enable_mpi $mpicc $enable_debug $enable_sse\
+-        CC="$cc" CFLAGS="$cflags $coptflags" \
+-        F77="$fc" FFLAGS="$fflags $foptflags" \
+-        FLIBS="$flibs_arch" \
+-        > ../fftw3_config.log 2>&1
+-      ncerror=$?
+-    fi
+-    if [ $ncerror -gt 0 ]; then
+-        echo "Error: FFTW configure returned $ncerror"
+-        echo "       FFTW configure failed! Check the fftw3_config.log file"
+-        echo "       in the $AMBERHOME/AmberTools/src directory."
+-        exit 1
++    if [ -n "$MKL_HOME" ]; then
++	echo
++	echo "Using FFTW3 from MKL"
++	echo
++	flibs_fftw3="-lfftw3xf_intel $flibs_mkl"
```

Comment:
This line is an issue if one uses GCC + OpenMPI + MKL. I suggest swapping these two conditions (i.e. have `$EBROOTFFTW` take precedence over `$MKL_HOME`. If one has `FFTW` loaded, it should be used rather than MKL's.

Reply:
Yeah, that makes sense. I'll change the patch and make some tests.

Reply:
Have you tested if it works for you?